src/jit/morph.cpp

```diff
@@ -7654,6 +7663,11 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         return result;
     }
 
+    if (((call->gtFlags & GTF_CALL_NO_RETURN) != 0) && 
+        (call->TypeGet() == TYP_VOID))
+    {
+        fgRemoveRestOfBlock = true;
+    }
```

Comment:
Somebody else on the jit team can chime in, but there should be some way to indicate that the return value is not actually going to be produced, so that this restriction can be lifted.